src/sagemaker/payloads.py

```diff
@@ -0,0 +1,174 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Utilities related to payloads of pretrained machine learning models."""
+from __future__ import absolute_import
+
+import logging
+from typing import Dict, List, Optional
+
+from sagemaker.jumpstart import utils as jumpstart_utils
+from sagemaker.jumpstart import artifacts
+from sagemaker.jumpstart.constants import DEFAULT_JUMPSTART_SAGEMAKER_SESSION
+from sagemaker.jumpstart.payload_utils import PayloadSerializer
+from sagemaker.jumpstart.types import JumpStartSerializablePayload
+from sagemaker.session import Session
+
+
+logger = logging.getLogger(__name__)
+
+
+def retrieve_all_examples(
+    region: Optional[str] = None,
+    model_id: Optional[str] = None,
+    model_version: Optional[str] = None,
+    serialize: bool = False,
+    tolerate_vulnerable_model: bool = False,
+    tolerate_deprecated_model: bool = False,
+    sagemaker_session: Session = DEFAULT_JUMPSTART_SAGEMAKER_SESSION,
+) -> Optional[List[str]]:
+    """Retrieves the compatible payloads for the model matching the given arguments.
+
+    Args:
+        region (str): The AWS Region for which to retrieve the Jumpstart model payloads.
+        model_id (str): The model ID of the JumpStart model for which to retrieve
+            the model payloads.
+        model_version (str): The version of the JumpStart model for which to retrieve
+            the model payloads.
+        serialize (bool): Whether to serialize byte-stream valued payloads by downloading
+            binary files from s3 and applying encoding, or to keep payload in pre-serialized
+            state. Set this option to False if you want to avoid s3 downloads or if you
+            want to inspect the payload in a human-readable form. (Default: False).
+        tolerate_vulnerable_model (bool): ``True`` if vulnerable versions of model
```

Comment:
nit: If you use code formatting (``) for True/False here, it should be used consistently